### PR TITLE
add tests for x86

### PR DESCRIPTION
**Pull request by DennyDai**
### add tests for x86

Tested on Ubuntu 18.04 LTS, binary compiled with gcc 7.5.0.

*It runs properly on the local x86 machine, but segfault will appear when using qemu-i386 to run the patched binary file.
**Add{RO, RW, RWInit}DataPatch will fail when [adding data of length 10,000.](https://github.com/DennyDai/patcherex/blob/feat/ezpz/tests/test_detourbackend_i386.py#L134)

### fix clang cmd

fix inconsistent types of variable compile_flags
https://github.com/angr/patcherex/blob/feat/ezpz/patcherex/patches.py#L73
https://github.com/angr/patcherex/blob/feat/ezpz/patcherex/utils.py#L740

### CODE REVIEW

**Comment by twizmwazin** [comment]
Should binaries be added here, rather than in the binaries repo? I see that many are already here, but in the interest of not having our repos blow up to hundreds of megabytes, it might be preferable to commit them there.

Also, I see nose being used. It would be preferable to use standard-library unittest or python's built in asserts in place of nose, as we are hoping to eventually drop that dependency.

**Comment by antoniobianchi333** [comment]
@twizmwazin so, would you move the binary to angr/binary?
I think it makes sense.

**Comment by DennyDai** [comment]
> Also, I see nose being used. It would be preferable to use standard-library unittest or python's built in asserts in place of nose, as we are hoping to eventually drop that dependency.

@twizmwazin 
Thank you for the advise, I've replaced the uses of nose with python's built in asserts statement.

**Comment by twizmwazin** [comment]
> 
> 
> @twizmwazin so, would you move the binaries to angr/binary?
> I think it makes sense.

I think moving binaries to the binaries repo would make sense, and would be consistent with other angr projects. To load them, we use the assumption that the binaries repo is cloned into the same folder patcherex is cloned into.

**Comment by DennyDai** [comment]
I've moved binaries to angr/binaries
https://github.com/angr/binaries/pull/54

**Comment by antoniobianchi333** [comment]
@twizmwazin do you think this is good to be merged now?

**Comment by twizmwazin** [comment]
A few notes:
- There is a print statement in there that should be removed
- Unless there is a reason to go against convention, test names should ideally be snake_case
- The run_all function can likely be replaced with `unittest.main()`
- pep8 dictates that there should be two newlines between module-level imports/functions/statements

With those minor tweaks, lgtm

**Comment by DennyDai** [comment]
@twizmwazin Thank you for your review. I've fixed them.

File: patcherex/utils.py [hunk 1]
```diff
@@ -748,8 +748,9 @@ def compile_c(code, optimization='-Oz', name_map=None, compiler_flags=("-m32",))
         fp.write(code)
         fp.close()
 
-        res = exec_cmd(("clang -nostdlib -mno-sse -ffreestanding %s -o %s -c %s " \
-                        % (optimization, object_fname, c_fname)) + " ".join(compiler_flags), shell=True)
+        print()
```
Comment (twizmwazin) [inline patcherex/utils.py:751]: missed this one print, then this should be merged

Comment (DennyDai) [reply patcherex/utils.py]: Sorry my bad! now fixed

**Pull request merged**